### PR TITLE
@kanaabe => News Date headers

### PR DESCRIPTION
**Pull request by eessex**
- Adds NewsDateHeader component to render styled article date or 'today'
- Replaces NavNews date with NewsDateHeader component
- Adds NewsDateDivider, for displaying date between articles in list
- Minor updates to NavNews styling

screen shot 2018-03-28 at 9 53 45 am
screen shot 2018-03-28 at 9 56 35 am
screen shot 2018-03-28 at 9 57 48 am

### CODE REVIEW

File: src/Components/Publishing/News/NewsDateHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  date: string
+}
+
+export const NewsDateHeader: React.SFC<Props> = props => {
+  const { date } = props
+  const today = new Date()
+  const isToday =
+    moment(date).format("MMM D, YYYY") === moment(today).format("MMM D, YYYY")
+  const hasYear = moment(date).format("YYYY") !== moment(today).format("YYYY")
+  const format = hasYear ? "MMM D, YYYY" : "MMM D"
```
Comment (kanaabe) [inline src/Components/Publishing/News/NewsDateHeader.tsx]: Nice, could we move these to https://github.com/artsy/reaction/blob/master/src/Components/Publishing/Constants.ts#L71? 

I know they're not used in other places atm, but it helps to quickly understand what _kinds_ of dates we use and could used to advocate for simplifying them.

**Comment by kanaabe** [comment]
Awesome, LGTM. Had one comment about moving the date formatting to Constants but feel free to use your own discretion

**Comment by kanaabe** [comment]
Oh also one other thought: would love to see screenshots/gif! Of course not necessary 😄

File: src/Components/Publishing/Constants.ts [hunk 2]
```diff
@@ -79,17 +84,25 @@ export const getDate = (date, format: DateFormat = "default") => {
...
         : moment(date)
             .tz("America/New_York")
             .format("MMM D, YYYY")
-
       const time = moment(date)
         .tz("America/New_York")
         .format("h:mm a")
       return `${day} at ${time}`
+    case "news":
+      return isToday
```
Comment (eessex) [inline src/Components/Publishing/Constants.ts]: "news" is maybe not the best in the context of the other types here, open to suggestions -- nothing stood out immediately.

**Pull request merged**